src/web/views/Content/Select/index.jsx

```diff
@@ -6,129 +6,252 @@ import axios from 'axios'
+        setImmediate(() => moveCursorToEnd(this.searchInput))
+      })
+
+      window.addEventListener('keyup', this.handleChangeActiveItem)
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    const { categories } = newProps
+    if (!categories || !categories.length) {
+      return
     }
+    if (this.state.step !== formSteps.INITIAL) {
+      return
+    }
+    this.setState({
+      step: categories.length > 1 ? formSteps.PICK_CATEGORY : formSteps.MAIN
+    })
   }
 
   componentWillUnmount() {
     delete window.botpress.pickContent
   }
 
+  searchContentItems() {
+    return this.props.fetchContentItemsRecent({
+      count: SEARCH_RESULTS_LIMIT,
+      searchTerm: this.state.searchTerm,
+      categoryId: this.state.categoryId || 'all'
+    })
+  }
+
   handleChangeActiveItem = e => {
     const index = this.state.activeItemIndex
     if (e.key === 'ArrowUp') {
       this.setState({ activeItemIndex: index > 0 ? index - 1 : index })
     } else if (e.key === 'ArrowDown') {
-      this.setState({ activeItemIndex: index < 4 ? index + 1 : index })
+      this.setState({ activeItemIndex: index < SEARCH_RESULTS_LIMIT - 1 ? index + 1 : index })
     } else if (e.key === 'Enter') {
       this.handlePick(this.props.contentItems[this.state.activeItemIndex])
     }
   }
 
-  search = event => {
-    if (['ArrowUp', 'ArrowDown'].includes(event.key)) {
+  onSearchChange = event => {
+    const newSearchTerm = event.target.value
+    const { searchTerm } = this.state
+    if (newSearchTerm === searchTerm) {
       return
     }
-
-    this.setState({ searchTerm: event.target.value })
-    this.props
-      .fetchContentItemsRecent({ searchTerm: event.target.value })
-      .then(() => this.setState({ activeItemIndex: 0 }))
+    this.setState({ searchTerm: newSearchTerm }, () => {
+      this.searchContentItems().then(() => this.setState({ activeItemIndex: 0 }))
+    })
   }
 
   handleCreate = () => {
     this.props
-      .upsertContentItems({ categoryId: this.state.category.id, formData: this.state.contentToEdit })
-      .then(() =>
-        Promise.all([
-          this.props.fetchContentItemsRecent({ searchTerm: this.state.searchTerm }),
-          this.props.fetchContentItemsCount()
-        ])
-      )
-      .then(() => this.setState({ category: null, contentToEdit: null }))
+      .upsertContentItem({
+        categoryId: this.state.newItemCategory.id,
+        formData: this.state.newItemData
+      })
+      .then(() => Promise.all([this.searchContentItems(), this.props.fetchContentItemsCount()]))
+      .then(this.resetCreateContent)
   }
 
   handlePick(item) {
-    this.setState({ show: false })
     this.callback(item)
-    window.onkeyup = null
+    this.onClose()
   }
 
   handleFormEdited = data => {
-    this.setState({ contentToEdit: data })
+    this.setState({ newItemData: data })
+  }
+
+  resetCreateContent = () => {
+    this.setState({ newItemCategory: null, newItemData: null, searchTerm: null })
```

Comment:
Note for myself: need to fix this, only reset the search term if the content was created, not form cancelled